modules/configuration/src/test/java/org/apache/ignite/internal/configuration/RenamedConfigurationTest.java

```diff
@@ -0,0 +1,279 @@
+import org.apache.ignite.configuration.annotation.PolymorphicConfig;
+import org.apache.ignite.configuration.annotation.PolymorphicConfigInstance;
+import org.apache.ignite.configuration.annotation.PolymorphicId;
+import org.apache.ignite.configuration.annotation.PublicName;
+import org.apache.ignite.configuration.annotation.Value;
+import org.apache.ignite.internal.configuration.storage.TestConfigurationStorage;
+import org.apache.ignite.internal.configuration.validation.TestConfigurationValidator;
+import org.apache.ignite.internal.manager.ComponentContext;
+import org.apache.ignite.internal.testframework.BaseIgniteAbstractTest;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RenamedConfigurationTest extends BaseIgniteAbstractTest {
+    private static final String OLD_VALUE_NAME = "oldName";
+    private static final String OLD_INNER_NAME = "oldInnerName";
+    private static final String SECOND_OLD_INNER_NAME = "secondOldInnerName";
+    private static final String OLD_LIST_NAME = "listOldName";
+    private static final String POLYMORPHIC_TYPE = "polymorphicType";
+    private static final String OLD_POLYMORPHIC_NAME = "oldPolymorphicName";
+
+    private static final ConfigurationTreeGenerator OLD_GENERATOR = new ConfigurationTreeGenerator(
+            Set.of(RenamedTestOldConfiguration.KEY),
+            Set.of(),
+            Set.of(RenamedPolymorphicInstanceOldConfigurationSchema.class)
+    );
+
+    private static final ConfigurationTreeGenerator NEW_GENERATOR = new ConfigurationTreeGenerator(
+            Set.of(RenamedTestNewConfiguration.KEY),
+            Set.of(),
+            Set.of(RenamedPolymorphicInstanceNewConfigurationSchema.class)
+    );
+    public static final String OLD_DEFAULT = "oldDefault";
+
+    private final TestConfigurationStorage storage = new TestConfigurationStorage(LOCAL);
+
+    private ConfigurationRegistry registry;
+
+    @AfterAll
+    public static void afterAll() {
+        OLD_GENERATOR.close();
+    }
+
+    @BeforeEach
+    void setUp() {
+        registry = startRegistry(RenamedTestOldConfiguration.KEY, OLD_GENERATOR);
+
+        stopRegistry(registry);
+
+        registry = startRegistry(RenamedTestNewConfiguration.KEY, NEW_GENERATOR);
+    }
+
+    @AfterEach
+    void tearDown() {
+        stopRegistry(registry);
+    }
+
+    @Test
+    public void testLegacyNameIsRecognisedOnStartup() {
+        // Default value was set when registry was started with old configuration.
+        assertThat(
+                registry.getConfiguration(RenamedTestNewConfiguration.KEY).newInnerName().newName().value(),
+                Matchers.equalTo(OLD_DEFAULT)
+        );
+
+        assertThat(storage.readLatest(String.format("key.%s.%s", OLD_INNER_NAME, OLD_VALUE_NAME)), willBe(nullValue()));
```

Comment:
Yeah, I'd prefer a hard-coded constant, to be honest with you. But I'm not insisting